sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -430,52 +430,63 @@ internal class QProductCenterManager internal constructor(
         purchaseHistoryRecords: List<PurchaseHistory>,
         restoreError: QonversionError
     ) {
-        val launchResult = launchResultCache.getLaunchResult() ?: run {
+        val productPermissions = launchResultCache.getProductPermissions() ?: run {
             failLocallyGrantingRestorePermissionsWithError(
                 launchError ?: QonversionError(QonversionErrorCode.LaunchError)
             )
             return
         }
 
-        launchResultCache.productPermissions?.let {
-            val permissions = grantPermissionsAfterFailedRestore(
-                purchaseHistoryRecords,
-                launchResult.products.values,
-                it
+        val products = launchResultCache.getActualProducts() ?: run {
+            failLocallyGrantingRestorePermissionsWithError(
+                launchError ?: QonversionError(QonversionErrorCode.LaunchError)
```

Comment:
Let's rewrite it in a simple way - first get both fields, then check both on null and then once fail if any of them is null